File: lib/modyilz/dayl_modyil.dart
import 'package:flutter/material.dart';
import '../modalz/angol_modalz.dart';
import '../modalz/kepad_konfeg.dart';
import '../yutelez/heksagon_djeyometre.dart';
import '../wedjets/heksagon_wedjet.dart';
import '../wedjets/sentir_mod_wedjet.dart';
import '../wedjets/enir_renq_wedjet.dart';

class DaylModyil extends StatelessWidget {
  final HeksagonDjeyometre geometry;
  final List<ModyilDeyda> modyilz;
  final Function(int) onToggleModule;

  const DaylModyil({
    super.key,
    required this.geometry,
    required this.modyilz,
    required this.onToggleModule,
  });

  @override
  Widget build(BuildContext context) {
    final daylModule =
        modyilz.firstWhere((m) => m.id == 'dayl', orElse: () => modyilz.first);

    return LayoutBuilder(
      builder: (context, constraints) {
        final stackWidth = constraints.maxWidth;
        final stackHeight = constraints.maxHeight;
        return Stack(
          alignment: Alignment.center,
          children: [
            // Center hexagon
            SentirModWedjet(
              label: daylModule.neym,
              geometry: geometry,
              backgroundColor: daylModule.kulor,
              textColor: KepadKonfeg.getComplementaryColor(daylModule.kulor),
              onTap: () => onToggleModule(daylModule.pozecon),
            ),
            // Inner ring layout
            EnirRenqWedjet(
              geometry: geometry,
              stackWidth: stackWidth,
              stackHeight: stackHeight,
              children: _innerRingWidgets,
            ),
          ],
        );
      },
    );
  }

  List<Widget> get innerRingWidgets => _innerRingWidgets;

  List<Widget> get _innerRingWidgets {
    final innerRingModules = modyilz.where((m) => m.id != 'dayl').toList();
    return innerRingModules.map((module) {
      return HeksagonWedjet(
        label: module.neym,
        backgroundColor: module.kulor,
        textColor: KepadKonfeg.getComplementaryColor(module.kulor),
        size: geometry.heksWidlx,
        isPressed: module.ezAktiv,
        rotationAngle: geometry.roteyconAngol,
        onTap: () => onToggleModule(module.pozecon),
      );
    }).toList();
  }
}

File: composeApp/src/commonMain/kotlin/modyilz/DaylModyil.kt
package modyilz

import androidx.compose.foundation.layout.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import yuteledez.HeksagonDjeyometre
import modalz.ModyilDeyda
import modalz.KepadKonfeg
import wedjets.SentirModWedjet
import wedjets.EnirRenqWedjet
import wedjets.HeksagonWedjet
import wedjets.HeksagonGred
import wedjets.GredItem

@Composable
fun DaylModyil(
    geometry: HeksagonDjeyometre,
    modyilz: List<ModyilDeyda>,
    onToggleModule: (Int) -> Unit,
    onMoveModule: (Int, Int) -> Unit,
    onCopyToEmpty: (Int, Int) -> Unit,
    onMoveToCenter: (Int) -> Unit,
    stackWidth: Dp,
    stackHeight: Dp
) {
    val daylModule = modyilz.find { it.id == "dayl" } ?: modyilz.first()
    val gredItems = modyilz.filter { it.id != "dayl" }.map { mod ->
        GredItem(
            index = mod.pozecon - 1,
            label = mod.neym,
            color = mod.kulor,
            isFolder = true,
            deyda = mod
        )
    }

    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
        HeksagonGred(
            geometry = geometry,
            items = gredItems,
            centerLabel = daylModule.neym,
            centerColor = daylModule.kulor,
            onMove = onMoveModule,
            onCopyToEmpty = onCopyToEmpty,
            onMoveToCenter = onMoveToCenter,
            onDropOnFolder = { from, to -> },
            modifier = Modifier.fillMaxSize(),
            onTap = onToggleModule
        )
    }
}
